lisa-utils/src/main/scala/lisa/utils/LisaException.scala

```diff
@@ -63,4 +66,14 @@ object UserLisaException {
   class UserParsingException(val parsedString: String, errorMessage: String)(using line: sourcecode.Line, file: sourcecode.File) extends UserLisaException(errorMessage) {
     def showError: String = ""
   }
+  class UserInvalidDefinitionException(val symbol: String, errorMessage: String)(using line: sourcecode.Line, file: sourcecode.File) extends UserLisaException(errorMessage) { // TODO refine
+    val showError: String = {
+      val source = scala.io.Source.fromFile(file.value)
+      val textline = source.getLines().drop(line.value - 1).next().dropWhile(c => c.isWhitespace)
+      source.close()
+      s"   Definition of $symbol at.(${file.value.split("/").last.split("\\\\").last}:${line.value}) is invalid:\n" +
+        "   " + Console.RED + textline + Console.BLACK + "\n\n" +
```

Comment:
Have we figured out a console color reset yet? Since setting console to black does change the color from default. I don't know off the top of my head if the scala lib has it, but might be better to use `Console.FOREGROUND` or equivalent.

Reply:
[Console.RESET does exist!](https://www.scala-lang.org/api/2.13.4/scala/Console$.html#RESET:String(%22\u001B[0m%22))

Reply:
We can add these to all the messages together at some point later.

Reply:
Cool! (for future pr)